File: 1/TreePrinter.py
# -*- coding:utf-8 -*-

# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class TreePrinter:
    def printTree(self, root):
        # write code here
        q = []
        res = []
        row = []
        if root == None:
            return res
        last = nlast = root
        q.append(root)
        while len(q):
            node = q.pop(0)
            row.append(node.val)
            if node.left != None:
                q.append(node.left)
                nlast = node.left
            if node.right != None:
                q.append(node.right)
                nlast = node.right
            if last == node:
                res.append(row)
                row = []
                last = nlast
        return res

File: 1/TreePrinter.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};*/

class TreePrinter {
public:
    vector<vector<int> > printTree(TreeNode* root) {
        queue<TreeNode*> tree;
        vector<vector<int>> result;
        vector<int> row;
        TreeNode* last = root;
        TreeNode* nlast = NULL;
        TreeNode* node = NULL;
        tree.push(root);
        
        while(!tree.empty()){
            node = tree.front();
            tree.pop();
            row.push_back(node->val);
            if(node->left){
                tree.push(node->left);
                nlast = node->left;
            }
            if(node->right){
                tree.push(node->right);
                nlast = node->right;
            }
            if(last == node){
                last = nlast;
                result.push_back(row);
                row.clear();
            }
        }
        return result;
    }
};